server/models/User.js

```diff
@@ -19,7 +19,15 @@ const userSchema = new mongoose.Schema({
   register_date: {
     type: Date,
     default: Date.now
-  }
+  },
+  followers: {
+    type: Array,
+    default: [],
+  },
+  followings: {
+    type: Array,
+    default: [],
+  },
```

Comment:
These should be references. 

(optional) Also there might be a better way to track follows without duplicating data on the follower and followed users. Maybe having a separate model with both ids and a method on the user to populate these two arrays when fetched.

Reply:
should I make somethings like a followRelationship model that holds the follower and following user IDs and add a reference to the user model to that
something like this:
```
const userSchema = new mongoose.Schema({
  ...
  followRelationships: {
    type:Array, 
    default:[],
    ref:'followrelationships'
  }
});
```

```
const FollowRelationship = new mongoose.Schema({
  follower: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'users'
  },
  following: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'users'
  }
});
```